pkg/cri/resource-manager/policy/builtin/memtier/pools.go

```diff
@@ -93,6 +99,35 @@ func (p *policy) buildPoolsByTopology() error {
 	return nil
 }
 
+// checkHWTopology verifies our otherwise implicit assumptions about the HW.
+func (p *policy) checkHWTopology() error {
+	// NUMA nodes (memory controllers) should not be shared by multiple sockets.
```

Comment:
we probably should take care about `die` part of topology reported by kernel

Reply:
Yes, but I think that warrants a separate PR. This PR just adds verification of the current implementations otherwise implicit HW assumptions.